### DISCUSSION TITLE
Query based on parent relationship attributes

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#2023] [Help] [answered]
Hello,

It seems that today I'm not that inspired, and formulating the question for big G isn't helping me. Maybe you guys can help me 🙏

So, let's say I have 3 entities, User, Post and Categories.

User has many posts
Post belongs to one Category,

How can I query the posts from a user in a category with a specific name (not id), like the following code:

```
const posts = await user
  .related('posts')
  .query()
  .where((query) => {
    if (params.categoryName) {
      // Something something here
    }
  })
  .orderBy('created_at', params.orderBy)
  .paginate(params.page, params.perPage)
```

**Comment by HappyLDE** [comment]
Or maybe just give category id or ids (where and whereIn) to posts 💯🥇

**Comment by iagobruno** [comment, marked as answer]
You can do something like this:

```ts
const category = await Category.findByOrFail('name', params.categoryName)

const posts = await category.related('posts')
  .query()
  .where('user_id', '=', params.userId)
  .orderBy('created_at', params.orderBy)
  .paginate(params.page, params.perPage)
```

**Reply by popovicieduard** [reply]
Actually, I found out that if I add a belongsTo relationship I can create a where clause on that, so this:
```
    if (params.categoryName) {
      // Something something here
    }
```
becomes:
```
if (params.categoryName) {
query
    .whereHas('category', (q) => {
        q.where({ name: params.categoryName })
    })
}
```
